### DISCUSSION TITLE
Using a BT controller within the bluetooth.io web interface

### DISCUSSION THREAD

**Discussion by rs1n** [darthcloud/BlueRetro#384] [Q&A] [answered]
I've grown to REALLY love BlueRetro and have already built two sets of dongles (DC and GC) with a third on the way (N64). I am still learning more about the software side (in particular the mappings), Right now, the directions say that the ESP32 must be in pairing mode (no controller connected) in order to use blueretro.io. So I usually disconnect everything before I tinker with preset mappings. However, I realized that for unlisted controllers, it would be much easier to populate the target buttons if I could just click on the corresponding source box and just push the button on the controller I wanted to use, rather than use a drop-down box with pre-made labels that might not make sense to someone trying to actually figure out if their controller/arcade stick even had such a button. Is this something that could be easily added?( For example, if my theoretical arcade stick just has 8 un-marked face buttons, it seems like I would have to do trial and error elimination in order to figure out what drop-down button option to use.) 

Another question: in the Advanced Config page, what is the purpose of the drop-down list of 12 devices? Is this where we are supposed to select the 12 types of devices we plan to use regardless of which player/port is assigned? Or is a means of assigning what device will be used for controller port 1 through 12 (where applicable)?

**Comment by darthcloud** [comment, marked as answer]
> I've grown to REALLY love BlueRetro and have already built two sets of dongles (DC and GC) with a third on the way (N64). I am still learning more about the software side (in particular the mappings), Right now, the directions say that the ESP32 must be in pairing mode (no controller connected) in order to use blueretro.io. 

Not sure where you read the pairing mode part, the only requirement is to not have a controller connected. (Having no controller will trigger pairing mode but it's not related to the config) (You could have a controller connect and the pairing mode running and you wouldn't be able to access config)

>So I usually disconnect everything before I tinker with preset mappings. However, I realized that for unlisted controllers, it would be much easier to populate the target buttons if I could just click on the corresponding source box and just push the button on the controller I wanted to use, rather than use a drop-down box with pre-made labels that might not make sense to someone trying to actually figure out if their controller/arcade stick even had such a button. Is this something that could be easily added?( For example, if my theoretical arcade stick just has 8 un-marked face buttons, it seems like I would have to do trial and error elimination in order to figure out what drop-down button option to use.)

The config isn't accessible while you got controller connected because advertising the config takes air time which in turn will affect controller latency.

It's something I thinking about right now actually. You would need to start advertising the menu manually. But a new page would act as a wizard and ask you to press the buttons one by one base on their location so that weird controller can have as low level mapping tied to their unique BDADDR. You would have the current mapping applying on top of that. This is probably not coming soon, probably v1.9+ stuff.

> 
> Another question: in the Advanced Config page, what is the purpose of the drop-down list of 12 devices? Is this where we are supposed to select the 12 types of devices we plan to use regardless of which player/port is assigned? Or is a means of assigning what device will be used for controller port 1 through 12 (where applicable)?

It's the later.

**Reply by rs1n** [reply]
> Not sure where you read the pairing mode part, the only requirement is to not have a controller connected. (Having no controller will trigger pairing mode but it's not related to the config) (You could have a controller connect and the pairing mode running and you wouldn't be able to access config)

I meant to say that everything had to be disconnected. In my mind I had associated that with being the same as pairing/syncing which is why I ended up writing erroneous things. (Still new to a lot of this stuff so your clarifications help a lot!) Thank you for the response; can't wait for new features and updates!
